tests/models/test_pyro.py

```diff
@@ -183,6 +184,32 @@ def _create_indices_adata_manager(adata: AnnData) -> AnnDataManager:
     return adata_manager
 
 
+def test_pyro_bayesian_regression_low_level():
+    use_gpu = int(torch.cuda.is_available())
+    adata = synthetic_iid()
+    adata_manager = _create_indices_adata_manager(adata)
+    train_dl = AnnDataLoader(adata_manager, shuffle=True, batch_size=128)
+    pyro.clear_param_store()
+    model = BayesianRegressionModule(in_features=adata.shape[1], out_features=1)
+    plan = LowLevelPyroTrainingPlan(model)
+    plan.n_obs_training = len(train_dl.indices)
+    trainer = Trainer(
+        gpus=use_gpu,
+        max_epochs=2,
+        callbacks=[PyroModelGuideWarmup(train_dl)],
```

Comment:
Is it essential to provide a dataloader like `PyroModelGuideWarmup(train_dl)`?

Reply:
yes

Reply:
actually no it's not required